### PR TITLE
INMP-144: Add new way for auth

### PR DESCRIPTION
**Pull request by Vallher**

### CODE REVIEW

File: appdynamics/provider.go [hunk 1]
```diff
@@ -21,13 +22,32 @@ func Provider() *schema.Provider {
...
+			"client_name":         {Type: schema.TypeString, Optional: true},
+			"client_secret":       {Type: schema.TypeString, Sensitive: true, Optional: true},
 		},
 		ConfigureFunc: func(data *schema.ResourceData) (interface{}, error) {
+			controllerBaseUrl := data.Get("controller_base_url").(string)
+			clientName := data.Get("client_name").(string)
+			clientSecret := data.Get("client_secret").(string)
+			token := data.Get("secret").(string)
+
+			if clientName == "" && clientSecret == "" && token == "" {
```
Comment (WRdbrulinski) [inline appdynamics/provider.go:36]: Are you sure this if makes sense?

File: appdynamics/provider.go [hunk 2]
```diff
@@ -21,13 +22,32 @@ func Provider() *schema.Provider {
 			"appdynamics_appd_service":     dataSourceAppdService(),
 		},
 		Schema: map[string]*schema.Schema{
-			"secret":              {Type: schema.TypeString, Sensitive: true, Required: true},
+			"secret":              {Type: schema.TypeString, Sensitive: true, Optional: true},
```
Comment (WRdbrulinski) [inline appdynamics/provider.go:25]: Not sure if both secret and client_secret set to optional is a good idea. Maybe additional field defining secret type?

Comment (Vallher) [reply appdynamics/provider.go:25]: Secret is to keep backward compatibility

File: appdynamics/provider.go (hunk 1 above)
Comment (Vallher) [reply appdynamics/provider.go:36]: Yes.

**Review by tkocemba — APPROVED** [review]
LGTM

**Review by mmisiorek-worldremit — APPROVED** [review]
LGTM

**Pull request merged**